syntax = "proto3";

package transport_catalogue_serialize;

message Stop {
    string name = 1;
    double lat = 2;
    double lng = 3;
    map<int32, double> distance = 4;
}

message StopList {
    repeated Stop stop = 1;
}

message Bus {
    string name = 1;
    bool is_roundtrip = 2;
    repeated int32 stop_id = 3;
}

message BusList {
    repeated Bus bus = 1;
}

message Point {
    double x = 1;
    double y = 2;
}

message Rgba {
    int32 red = 1;
    int32 green = 2;
    int32 blue = 3;
    double opacity = 4;
}

message Rgb {
    int32 red = 1;
    int32 green = 2;
    int32 blue = 3;
}

message Color {
oneof color_oneof {
    string name = 1;
    Rgba rgba = 2;
    Rgb rgb = 3;
  }
}

message RenderSettings {
    double width = 1;
    double height = 2;

    double padding = 3;

    double line_width = 4;
    double stop_radius = 5;

    int32 bus_label_font_size = 6;
    Point bus_label_offset = 7;

    int32 stop_label_font_size = 8;
    Point stop_label_offset = 9;

    Color underlayer_color = 10;
    double underlayer_width = 11;

    repeated Color color_palette = 12;
}

message Database {
    StopList stop_list = 1;
    BusList bus_list = 2;
    RenderSettings render_settings = 3;
}
